### PR TITLE
updated to latest version of actions for set-python and codecov

### LINKED ISSUE #2789: CI GH Actions updates
According to recent warnings on unit tests:
```
cpu-tests (ubuntu-latest, 3.9, pytorch)
Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: actions/setup-python@v2, codecov/codecov-action@v1
```
we need to update versions for `actions/setup-python@v2` and `codecov/codecov-action@v1`.

What to do:
- Find the latest version for `actions/setup-python@v2` and `codecov/codecov-action@v1`
- Update the version for all our GH actions workflows by searching in the codebase
- Please send a PR

### PR DESCRIPTION
**Pull request by RishiKumarRay**
Fixes #2789

Description: updated to latest version for actions/setup-python@v2 and codecov/codecov-action@v1

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: .github/workflows/docker-build.yml [hunk 1]
```diff
@@ -46,7 +46,7 @@ jobs:
         run: |
           sudo rm -rf "/usr/local/share/boost"
           sudo rm -rf "$AGENT_TOOLSDIRECTORY"
-      - uses: actions/setup-python@v2
+      - uses: actions/setup-python@v2.3.3
```
Comment (vfdev-5) [inline .github/workflows/docker-build.yml:49]: Let's use v4 instead. See https://github.com/actions/setup-python#basic-usage

File: .github/workflows/hvd-tests.yml [hunk 2]
```diff
@@ -78,7 +78,7 @@ jobs:
           bash tests/run_cpu_tests.sh
 
       - name: Upload coverage to Codecov
-        uses: codecov/codecov-action@v1
+        uses: codecov/codecov-action@v3.1.1
```
Comment (vfdev-5) [inline .github/workflows/hvd-tests.yml:81]: Here and elsewhere let's only keep major version:
```suggestion
        uses: codecov/codecov-action@v3
```

**Review by vfdev-5 — COMMENTED** [review]
Thanks for the PR @RishiKumarRay. I left few comment on how to update it.

File: .github/workflows/docker-build.yml (hunk 1 above)
Comment (RishiKumarRay) [reply .github/workflows/docker-build.yml]: @vfdev-5 btw when i checked the latest version it was showing 2.3.3 at marketplace but yeah i have done it to v4 
Screenshot from 2022-12-01 19-42-36

Comment (vfdev-5) [reply .github/workflows/docker-build.yml]: v2.3.3 seems like a backport of some newest features, https://github.com/actions/setup-python/releases

**Review by vfdev-5 — APPROVED** [review]
LGTM, thanks @RishiKumarRay !

**Pull request merged**